Date: Wed, 22 Jun 2005 08:02:03 +0200
From: Uros Bizjak <uros dot bizjak at kss-loka dot si>
Subject: [PATCH, i386]: Fix target/22134

Hello!

This patch fixes PR target/22134. It was a cut-n-pasto. Patch was tested on
i686-pc-linux-gnu and fixes a regression.

2005-05-22  Uros Bizjak  <uros@kss-loka.si>

	PR target/22134
	* config/i386/i386.md (fist<mode>2_with_temp splitter):
	Clobber memory operand, not scratch register.

Thanks,
Uros.

--- a/gcc/config/i386/i386.md	2005-06-22 07:07:28.000000000 +0200
+++ b/gcc/config/i386/i386.md	2005-06-22 07:07:45.000000000 +0200
@@ -16213,7 +16213,7 @@
   [(set (match_operand:X87MODEI12 0 "memory_operand" "")
 	(unspec:X87MODEI12 [(match_operand:XF 1 "register_operand" "")]
 	 UNSPEC_FIST))
-   (clobber (match_scratch 2 ""))]
+   (clobber (match_operand:X87MODEI12 2 "memory_operand" ""))]
   "reload_completed"
   [(set (match_dup 0) (unspec:X87MODEI12 [(match_dup 1)]
 		       UNSPEC_FIST))]
